cmd/exec.go

```diff
@@ -120,7 +126,19 @@ func runExecWithContext(ctx context.Context, params *exec.Params) error {
 		return fmt.Errorf("runtime error: %w", err)
 	}
 
-	<-ready
+	select {
+	case <-ctx.Done():
+		err := ctx.Err()
+		if err == context.DeadlineExceeded {
+			return fmt.Errorf("exec error: timed out before OPA was ready. This can happen when a remote bundle is malformed, or the timeout is set too low for normal OPA initialization")
+		} else {
+			// Note(philipc): Previously, exec would simply eat the context
+			// cancellation error. We now propagate that upwards to the caller.
+			return err
```

Comment:
This is arguably a "more correct" behavior than simply swallowing the error, as was done in the past.

Also, because this is a package-private function, it should be quite safe for us to change its behavior (the only affected downstream *should* be our own test suite)